Documentation/docs/onprem/configuration/configure-mail-server.md
---
description: To configure your mail serveur with Twake.
---

# 💌 Configure mail server

Here is the default mail block in the configuration file to edit \(from Parameters.php, see "[Configuration page](./)"\):

```text
"mail" => [
    "sender" => [
        "host" => "", //smtp server
        "port" => "",
        "username" => "",
        "password" => "",
        "auth_mode" => "plain" //plain, login, cram-md5, or null
    ],
    "from" => "noreply@twakeapp.com",
    "dkim" => [ //Optional, to avoid lost emails, configure your dns with dkim
        "private_key" => "",
        "domain_name" => '',
        "selector" => ''
    ],
    "twake_domain_url" => "https://twakeapp.com/",
    "from_name" => "Twake", //Server owner name
    "twake_address" => "Twake, 54000 Nancy, France", //Server owner address
    "template_dir" => "/src/Twake/Core/Resources/views/", //Must not be modified
],
```

⚠️ Once edited, don't forget to restart docker.

You can test the good behaviour of emails going into your account parameters, emails, add a secondary email. Or simply try to invite a user using its email.



Documentation/docs/onprem/configuration/custom-domain-and-https/apache2-configuration.md
---
description: "\U0001F64F From Dahpril (community) https://github.com/TwakeApp/Twake/issues/76"
---

# Apache2 configuration

#### Apache vhost

You need to create a vhost listening on port 443.

_For exemple with certbot :_

_1. Create a vhost listening on port 80 with ServerName equals to your custom domain. Don't define any document root. 2. Then, use certbot to get a certificate and automatically create your vhost listening on port 443_

```text
sudo certbot --apache --email your_email -d your_domain --agree-tos --redirect --noninteractive
```

#### Reverse proxy

You have now to configure your reverse proxy directive. Head up to your 443 vhost configuration file and paste those directives \(place them after server and ssl directives\) :

```text
RewriteEngine on
RewriteCond ${HTTP:Upgrade} websocket [NC]
RewriteCond ${HTTP:Connection} upgrade [NC]
RewriteRule .* "wss://127.0.0.1:8000/$1" [P,L]

ProxyRequests off

<Location />
    ProxyPass http://127.0.0.1:8000/
    ProxyPassReverse http://127.0.0.1:8000/
    ProxyPreserveHost On
</Location>
<Location /socketcluster>
    ProxyPass ws://127.0.0.1:8000/socketcluster
    ProxyPassReverse ws://127.0.0.1:8000/socketcluster
    ProxyPreserveHost On
</Location>

<Proxy *>
    AllowOverride All
    Order allow,deny
    Allow from All
</Proxy>

RequestHeader set X-Forwarded-port "80"
```

Be careful to NOT type trailing slash for ws location \(it won't work\).

I'm not sure that all directives are needed, but this configuration works for me.

#### Configuring remoteip

You also need to configure remoteip mod from apache with this command :

```text
a2enmod remoteip
```

Then, edit /etc/apache2/conf-available/remoteip.conf to fit with this content :

```text
RemoteIPHeader X-Real-IP
RemoteIPTrustedProxy 127.0.0.1 ::1
```

You can now enable the configuration of remoteip and restart Apache :

```text
a2enconf remoteip
service apache2 restart
```

Return to the section "Configure domain name" of [Custom domain + HTTPS](./#configure-domain-name) page to continue the configuration.



changelog.md
# Twake 2022.Q4.1143

- Ability to invite automatically any user with same business email ✉️
- Workspace invitation improved 🎨
- Channel menu improved 🎨
- Fixes on Documents (for SaaS) 📁
- Forward messages to another channel ⤴️
- Set a channel as read-only 👀

# Twake 2022.Q4.1120

- Message delivery status 🟢
- New quote-reply feature ⤴️
- Notification when someone react to your message 👍
- New file preview and medias gallery with navigation, caption, and message context 🕶
- New badge notification system (mention, unread, etc)
- Digest notifications by email ✉️
- Ability to customize notification sound
- Improved user live status indicator
- See when user was last active on Twake ⏰
- Big simplification of backend 👩‍💻

# Twake 2022.Q3.1065

- File and medias channel gallery view 🎞
- New channel members invitation and management modal ⚙️
- Fixed connection status of users 🟢
- New messages search feature on mobile 🔎
- Fixed bugs and translation #2425 #2424 #2420 #2442 #2436 #2435

# Twake 2022.Q3.1050

- Open desktop app if present in the system
- Preview public channels before to join them
- Updated [Search anything on Twake](https://github.com/linagora/Twake/issues/1710)
- Started using Tailwindcss for future new design
- Started using Storybooks

# Twake 2022.Q2.975

### Backend

- [Implemented the list of uploaded in our backend](https://github.com/linagora/Twake/issues/2127)
- [Jump to a message](https://github.com/linagora/Twake/issues/1926)
- [Search for messages](https://github.com/linagora/Twake/issues/1710)
- Generate links previews
- Multiple bug fixes

### Bug fixed

- [Fixed thumbnail rotatation when uploaded from mobile](https://github.com/linagora/Twake/issues/2111)
- [Scrolling message feed isn't smooth](https://github.com/linagora/Twake/issues/2039)
- [Position of Close (x) buttons are not consistent](https://github.com/linagora/Twake/issues/2031)
- [Mention user by any namu or identifier](https://github.com/linagora/Twake/issues/2042)
- [Update workspace name reset the logo](https://github.com/linagora/Twake/issues/1950)
- [Empty channel isn't working](https://github.com/linagora/Twake/issues/2146)
- [On Twake desktop I have notifications only inside Twake window and when it's collapsed I don't see them](https://github.com/linagora/Twake/issues/2043)
- [Windows notification has no redirection](https://github.com/linagora/Twake/issues/125)
- [Clicking on push notification does not open the right message](https://github.com/linagora/Twake/issues/396)

# Twake 2022.Q2.930

### Backend

- Refactoring of how we get services in our backend

### Bug fixed

- https://github.com/linagora/Twake/pull/2092

# Twake 2022.Q2.910

### Messages

- Copy paste images to message editor fixed
- Large files preview is now available
- Mentions are back in messages
- Giphy, n8n and Jitsi are back

### General

- New workspaces parameters page view
- Ability to create and manage plugins from Twake

### Backend

- New backend routes to get files and pinned messages of a channel
- New search messages routes

### Bug fixed

- Fixed thumbnail generation
- https://github.com/linagora/Twake/issues/2083
- https://github.com/linagora/Twake/issues/2091
- https://github.com/linagora/Twake/issues/2095
- https://github.com/linagora/Twake/issues/2037
- https://github.com/linagora/Twake/issues/2036
- https://github.com/linagora/Twake/issues/1918
- https://github.com/linagora/Twake/issues/2102

# Twake 2022.Q1.890

### Messages

- Shows that the last message was deleted on mobile channel list view
- Now we can cancel failed file uploads
- Add spell check to message editor

### General

- Save the user last visited companies to open them again after logout
- Fixed magic links
- Show a popup to decide to continue on web or open the mobile app
- Notice users that the company reached the maximum number of users
- Fixed some issues with workspace invitation
- Minor bugs fixed

# Twake 2022.Q1.874

### Messages

- Faster loading of message feed
- Fix message input issues

### General

- Reduce magic links invite urls
- Fix guest popup not visible on paid plans
- Fix company owners and admins not able to manage workspaces
- Fix channels members counters
- Migrate channels to recoil state management for more stable channels (favorite, rename, create, join etc)

# Twake 2021.Q4.860

### Messages

- Improved message top right menu
- Fixed message sending / edition / reaction / pinned etc, now instant ⚡️
- User availability indicator 🟢
- User is writing indicator

### Workspace parameters

- Improved integrations management page 🧹
- Improved workspace users management page 🧹
- Improved workspace preferences management page 🧹

### General

- A lot of bug fixed, more to come on the 22 of January 🐞
- Improved Twake loading time (still in progress) ⏰
- New state management ( recoiljs.org ) on everything except channels and channels members
- Everything except Drive Tasks and Calendar was migrated to node backend 🧹
- Twake is now working with a simplified docker-compose with only node + mongodb. Note that this works only with messages (no Drive, Calendar or Tasks yet) 🚀

# Twake 2021.Q3.640

### Messages

- New message WYSIWYG editor 🎨
- Added limit to message size and to number of attached files
- Deleted messages are now showed as "Deleted" but don't disappear
- Limit number of users in the same direct channel
- You can now set multiple reactions to a single message 🌟👍🔥

### General

- Twake is now on [Weblate](https://hosted.weblate.org/projects/twake/) 🌎
- Multiple bugs fixes

# Twake 2021.Q2.550

### Messages

- Fix missing notification when a first direct channel message is sent
- Now using the new nodejs backend (need to run `sudo docker-compose run node node bin/twake-cli migration message` after update). PHP endpoints (`/ajax/discussion/...`) are still working but are depreciated.

### General

- Update oidc login workflow for Console connectivity

### Technical changes


twake/default-configuration/backend/Parameters.php
<?php

namespace Configuration;

class Parameters extends \Common\Configuration
{

    public $configuration = [];

    public function __construct()
    {
        $this->configuration = [
            "env" => [
              "type"  =>  "prod",
              "timer" => false,
              "admin_api_token" => "",
              "licence_key" => "",
              "standalone" => true,
              //"frontend_server_name" => "http://localhost:8000/", // define only if needed
              "server_name" => "http://localhost:8000/",
              "internal_server_name" => "http://nginx/"
            ],
            "jwt" => [
              "secret" => "supersecret",
              "expiration" => 60*60, //1 hour
              "refresh_expiration" => 60*60*24*31 //1 month
            ],
            "node" => [
                "api" => "http://node:3000/private/",
                "secret" => "api_supersecret"
            ],
            "db" => [
                "driver" => "pdo_cassandra",
                "host" => "scylladb",
                "port" => 9042,
                "dbname" => "twake",
                "user" => "root",
                "password" => "root",
                "encryption_key" => "ab63bb3e90c0271c9a1c06651a7c0967eab8851a7a897766",
                "replication" => "{'class': 'SimpleStrategy', 'replication_factor': '1'}"
            ],
            "es" => [
              "host" => false
            ],
            "queues" => [
                "rabbitmq" => [
                    "use" => true,
                    "host" => "rabbitmq",
                    "port" => 5672,
                    "username" => "guest",
                    "password" => "guest",
                    "vhost" => false
                ]
            ],
            "storage" => [
              "drive_previews_tmp_folder" => "/tmp/",
              "drive_tmp_folder" => "/tmp/",
              "drive_salt" => "SecretPassword",
              "providers" => [
                [
                    "label" => "someOpenStack",
                    "type" => "openstack",
                    "use" => false,
                    "project_id" => "",
                    "auth_url" => "https//auth.cloud.ovh.net/v3",
                    "buckets_prefix" => "",
                    "disable_encryption" => false,
                    "buckets" => [
                        "fr" => [
                            "public" => "",
                            "private" => "",
                            "region" => "SBG5"
                        ]
                    ],
                    "user" => [
                        "id" => "",
                        "password" => "",
                        "domain_name" => "default"
                    ]
                ],
                [
                    "label" => "someS3",
                    "type" => "S3",
                    "base_url" => "http://127.0.0.1:9000",
                    "use" => false,
                    "version" => "latest",
                    "disable_encryption" => false,
                    "buckets_prefix" => "dev.",
                    "buckets" => [
                        "fr" => "eu-west-3"
                    ],
                    "credentials" => [
                        "key" => "",
                        "secret" => ""
                    ]
                ],
                [
                    "label" => "someLocal",
                    "type" => "local",
                    "use" => true,
                    "disable_encryption" => false,
                    "location" => "../drive/",
                    "preview_location" => "../web/medias/",
                    "preview_public_path" => "/medias/"
                ],
              ],
            ],
            "mail" => [
                "sender" => [
                    "host" => "",
                    "port" => "",
                    "username" => "",
                    "password" => "",
                    "auth_mode" => "plain"
                ],
                "template_dir" => "/src/Twake/Core/Resources/views/",
                "twake_domain_url" => "https://twakeapp.com/",
                "from" => "noreply@twakeapp.com",
                "from_name" => "Twake",
                "twake_address" => "Twake, 54000 Nancy, France",
                "dkim" => [
                    "private_key" => "",
                    "domain_name" => '',
                    "selector" => ''
                ],
                "mailjet"=> [
                  "contact_list_subscribe"=> false,
                  "contact_list_newsletter"=> false
                ]
            ],
            "push_notifications" => [
                "apns_certificate" => __DIR__ . "/certs/apns_prod.pem",
                "firebase_api_key" => "KEY",
            ],

            //Defaults values for all clients but editable in database
            "defaults" => [
              "applications" => [
                "twake_drive" => [ "default" => true ], //False to not install
                "twake_calendar" => [ "default" => true ],
                "twake_tasks" => [ "default" => true ],
                "connectors" => [
                  "jitsi" => [ "default" => true ],
                ]
              ],
              "connectors" => [
              ],
              "branding" => [
                "name" => "Twake",
                "enable_newsletter" => false,
                /*
                "header" => [
                  "logo" => 'https://openpaas.linagora.com/images/white-logo.svg',
                  "apps" => [
                    [
                      "name"=> 'Accueil',
                      "url"=> 'https://openpaas.linagora.com/',
                      "icon"=> 'https://openpaas.linagora.com/images/application-menu/home-icon.svg',
                    ],
                    [
                      "name"=> 'Inbox',
                      "url"=> 'https://openpaas.linagora.com/#/unifiedinbox/inbox',
                      "icon"=> 'https://openpaas.linagora.com/unifiedinbox/images/inbox-icon.svg',
                    ],
                    [
                      "name"=> 'Calendrier',
                      "url"=> 'https://openpaas.linagora.com/#/calendar',
                      "icon"=> 'https://openpaas.linagora.com/calendar/images/calendar-icon.svg',
                    ],
                    [
                      "name"=> 'Contacts',
                      "url"=> 'https://openpaas.linagora.com/#/contact/addressbooks/',
                      "icon"=> 'https://openpaas.linagora.com/contact/images/contacts-icon.svg',
                    ],
                  ],
                ],
                "style" => [
                  "color" => '#2196F3',
                  "default_border_radius" => '2',
                ],
                "link" => "https://open-paas.org/",
                "logo" => "https://open-paas.org/wp-content/uploads/2019/10/openpaas.png"
                */
              ],
              "auth" => [
                  "internal" => [
                      "use" => true,
                      "disable_account_creation" => false,
                      "disable_email_verification" => true
                  ],
                  "openid" => [
                      "use" => false,
                      "provider_uri" => 'https://auth0.com',
                      "client_id" => '',
                      "client_secret" => '',
                      //"disable_logout_redirect" => false
                      "provider_config" => [
                        //token_endpoint
                        //token_endpoint_auth_methods_supported
                        //userinfo_endpoint
                        //end_session_endpoint
                        //authorization_endpoint
                      ]
                  ],
                  "cas" => [
                      "use" => false,
                      "base_url" => '',
                      "email_key" => '',
                      "lastname_key" => '',
                      "firstname_key" => ''
                  ],
                  "console" => [
                    "use" => false
                  ]
              ]
            ],
        ];
    }

}
